app/views/admin/enterprises/form/_vouchers.html.haml

```diff
@@ -20,13 +21,23 @@
         %tr
           %td= voucher.code
           %td= voucher.display_value  
+          %td= f.check_box :voucher_ids, { :multiple => true }, voucher.id, nil
+          /%td
+          /%td
+          /%td
+          /%td
+          /%td
+      - @enterprise.vouchers.only_deleted.each do |voucher|
```

Comment:
Instead of repeating the table row definition I would change the database query:
```suggestion
      - @enterprise.vouchers.with_deleted.order(:deleted_at, :code).each do |voucher|
```